src/index.js

```diff
@@ -130,13 +130,12 @@ export let ContextTree = _.curry(
 
       // make all other nodes filter only
       if (path) {
-        Tree.walk(node => {
-          node.filterOnly = true
+        Tree.walk((node, index, parents) => {
+          let nodePath = [..._.map('key', _.reverse(parents)), node.key]
```

Comment:
Let’s just add a quick comment explaining that this is about marking everything that isn’t the node or it’s children as filter only